src/api/controllers/resetPassword.controller.js

```diff
@@ -0,0 +1,93 @@
+import { getUser } from "../services/auth.service";
+import { sendEmail, sendResetPasswordEmail } from "../utils/email";
+import { hashPassword } from "../utils/hash-password";
+import { generateResetToken } from "../utils/reset-token";
+
+export const resetPasswordEmail = async (req, res) => {
+  const { email } = req.body;
+
+  try {
+    const user = await getUser(email);
+    if (!user) {
+      return res.status(400).json({
+        success: false,
+        message: "User not found",
+      });
+    }
+  
+    const {resetToken, resetTokenExpiration} = generateResetToken()
+  
+    user.resetPasswordToken = resetToken
+    user.resetPasswordExpires = resetTokenExpiration
+
+    await user.save()
+    sendEmail('Password Reset Request', email, `You are receiving this email because you (or someone else) has requested the reset of the password for your account.<br><br>
+    Please click on the following link, or paste this into your browser to complete the process:<br><br>
+    <a href="https://phantom-techbandit.netlify.app/reset-password/${resetToken}">Reset Password</a><br><br>
+    This link is valid for 1 hour.<br><br>
+    If you did not request this, please ignore this email and your password will remain unchanged.<br>`)
+
+    return res.status(200).json({
+        success: true,
+        message: "reset password email sent",
+    })
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({
+        success: true,
+        message: "reset password email fail",
+    })
+  }
+};
+
+const findUserByResetToken = async (token) => {
+  try {
+    const user = await User.findOne({
+      where: {
+          resetPasswordToken: token,
+      },
+    });
+    return user;
+  } catch (error) {
+      return null
+  }
+}
+
+export const resetPassword = async (req, res) => {
+  const resetToken = req.params.resetToken
+  const { newPassword } = req.body;
+
+  try {
+    const user = await findUserByResetToken(resetToken);
+    if (!user) {
+        return res.status(404).json({
+            success: true,
+            message: "invalid or expired token",
+        })
+    }
+
+    const expirationDate = user.dataValues.resetPasswordExpires;
+
+    if (expirationDate.getTime() < Date.now()) {
+      return res.status(404).json({
+          success: true,
+          message: "invalid or expired token",
+      })
+    }
```

Comment:
add in middleware like above 
- firstly check if token doesn't expired
- secondly check if expiration data is valid